packages/flutter/lib/src/material/theme_data.dart

```diff
@@ -667,53 +714,120 @@ class ThemeData {
+        defaultValue: defaultData.dividerColor));
+    description.add(new DiagnosticsProperty<Color>('highlightColor', highlightColor,
+        defaultValue: defaultData.highlightColor));
+    description.add(new DiagnosticsProperty<Color>('splashColor', splashColor,
+        defaultValue: defaultData.splashColor));
+    description.add(new DiagnosticsProperty<Color>('selectedRowColor', selectedRowColor,
+        defaultValue: defaultData.selectedRowColor));
+    description.add(new DiagnosticsProperty<Color>('unselectedWidgetColor', unselectedWidgetColor,
+        defaultValue: defaultData.unselectedWidgetColor));
+    description.add(new DiagnosticsProperty<Color>('disabledColor', disabledColor,
+        defaultValue: defaultData.disabledColor));
+    description.add(new DiagnosticsProperty<Color>('buttonColor', buttonColor,
+        defaultValue: defaultData.buttonColor));
+    description.add(new DiagnosticsProperty<Color>('secondaryHeaderColor', secondaryHeaderColor,
+        defaultValue: defaultData.secondaryHeaderColor));
+    description.add(new DiagnosticsProperty<Color>('textSelectionColor', textSelectionColor,
+        defaultValue: defaultData.textSelectionColor));
+    description.add(new DiagnosticsProperty<Color>(
+        'textSelectionHandleColor', textSelectionHandleColor,
+        defaultValue: defaultData.textSelectionHandleColor));
+    description.add(new DiagnosticsProperty<Color>('backgroundColor', backgroundColor,
+        defaultValue: defaultData.backgroundColor));
+    description.add(new DiagnosticsProperty<Color>('dialogBackgroundColor', dialogBackgroundColor,
+        defaultValue: defaultData.dialogBackgroundColor));
+    description.add(new DiagnosticsProperty<Color>('indicatorColor', indicatorColor,
+        defaultValue: defaultData.indicatorColor));
+    description.add(new DiagnosticsProperty<Color>('hintColor', hintColor,
+        defaultValue: defaultData.hintColor));
+    description.add(new DiagnosticsProperty<Color>('errorColor', errorColor,
+        defaultValue: defaultData.errorColor));
+    description.add(new DiagnosticsProperty<ButtonThemeData>('buttonTheme', buttonTheme));
+    description.add(new DiagnosticsProperty<TextTheme>('textTheme', textTheme));
+    description.add(new DiagnosticsProperty<TextTheme>('primaryTextTheme', primaryTextTheme));
+    description.add(new DiagnosticsProperty<TextTheme>('accentTextTheme', accentTextTheme));
+    description.add(new DiagnosticsProperty<InputDecorationTheme>(
+        'inputDecorationTheme', inputDecorationTheme));
+    description.add(new DiagnosticsProperty<IconThemeData>('iconTheme', iconTheme));
+    description.add(new DiagnosticsProperty<IconThemeData>('primaryIconTheme', primaryIconTheme));
+    description.add(new DiagnosticsProperty<IconThemeData>('accentIconTheme', accentIconTheme));
+    description.add(new DiagnosticsProperty<SliderThemeData>('sliderTheme', sliderTheme));
```

Comment:
we should probably mark most of these such that they don't show up in the debugDumpApp() output, because it basically makes the output unreadable as it stands today.

Reply:
(also, minor nit: this would be more readable if all the lines were consistently wrapped or not wrapped)

Reply:
OK, in another PR I've reformatted these with dartfmt, but with a line length of 200.  That seems to keep the line breaks at bay.

Reply:
i was more worried about the problem of this all making debugDumpApp useless :-)

Reply:
Yeah, I was wondering about that too.  How does one mark them so they don't show up there?  I didn't see a switch on the DiagnosticsProperty object.  I added these because the previous toString was useless for debugging one of my tests.